File: coding/13th December 2024/Minimum difference Between Highest and Lowest of k scores.java
//Minimum difference Between Highest and Lowest of k scores
class Solution 
{
    public int minimumDifference(int[] nums, int k) 
    {
        int min = Integer.MAX_VALUE;
        Arrays.sort(nums);
        for(int i = 0; i < nums.length - k + 1; i++)
        {
            int j = i + k - 1;
            System.out.println(nums[i] + ' ' + nums[j]);
            int temp = nums[j] - nums[i];
            if(temp < min)
                min = temp;
        }
        return min;
    }
}

File: Coding in Python/25th January 2026/Minimum Difference Between Highest and Lowest of K Scores.py
#Minimum Difference Between Highest and Lowest of K Scores
class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:
        min1 = float('inf')
        nums.sort()
        for i in range(0, len(nums) - k + 1):
            j = i + k - 1
            temp = nums[j] - nums[i]
            if(temp < min1):
                min1 = temp
        return min1
